Source/DafnyCore/Resolver/PreType/PreTypeResolve.Expressions.cs

```diff
@@ -1085,8 +1085,11 @@ public partial class PreTypeResolver {
     /// <param name="allowMethodCall">If false, generates an error if the name denotes a method. If true and the name denotes a method, returns
     /// a MemberSelectExpr whose .Member is a Method.</param>
     /// <param name="complain"></param>
+    /// <param name="specialOpaqueHackAllowance">If "true", treats an expression "f" where "f" is an instance function, as "this.f", even though
+    /// there is no "this" in scope. This seems like a terrible hack, because it breaks scope invariants about the AST. But, for now, it's here
+    /// to mimic what the legacy resolver does.</param>
     Expression ResolveNameSegment(NameSegment expr, bool isLastNameSegment, List<ActualBinding> args,
-      ResolutionContext resolutionContext, bool allowMethodCall, bool complain = true) {
+      ResolutionContext resolutionContext, bool allowMethodCall, bool complain = true, bool specialOpaqueHackAllowance = false) {
```

Comment:
It would be nice to have this change in a separate PR, but I don't think it's a blocker.

Reply:
You're right, this change does not belong in this PR. I'm moving it to a separate one and am reverting `git-issue-4315.dfy` to just run with the old resolver.

Reply:
The separate PR is https://github.com/dafny-lang/dafny/pull/4733